File: src/csharp/Foundgine.Core/Semantic/Mutation/SemanticMutationField.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Mutation;

/// <summary>
/// A semantic field assignment. The field is identified semantically; physical
/// column mapping is deliberately outside this representation.
/// </summary>
public sealed record SemanticMutationField(
    FieldId Field,
    object? Value,
    SemanticMutationValueReference? Source = null);

File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/mutation/SemanticMutationField.java
package com.foundgine.core.semantic.mutation;

import com.foundgine.core.abstractions.FieldId;

/**
 * Semantic field assignment; physical column mapping is outside this
 * representation.
 */
public record SemanticMutationField(FieldId field, Object value, SemanticMutationValueReference source) {
	public SemanticMutationField {
		if (field == null)
			throw new NullPointerException("field");
	}

	public SemanticMutationField(FieldId field, Object value) {
		this(field, value, null);
	}
}
